internal/command/root.go

```diff
@@ -2,110 +2,143 @@ package command
 	"github.com/weaveworks/flintlock/pkg/log"
 )
 
-func NewRootCommand() (*cobra.Command, error) {
+const usage = `
+  __  _  _         _    _               _        _
+ / _|| |(_) _ __  | |_ | |  ___    ___ | | __ __| |
+| |_ | || || '_ \ | __|| | / _ \  / __|| |/ // _' |
+|  _|| || || | | || |_ | || (_) || (__ |   <| (_| |
+|_|  |_||_||_| |_| \__||_| \___/  \___||_|\_\\__,_|
+
+Create and manage the lifecycle of MicroVMs, backed by containerd
+`
+
+const configName = "config.yaml"
+
+func NewApp(out io.Writer) *cli.App {
 	cfg := &config.Config{}
+	// Append to the default template
+	cli.AppHelpTemplate = fmt.Sprintf(`%s
 
-	cmd := &cobra.Command{
-		Use:   "flintlockd",
-		Short: "The flintlock API",
-		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
-			flags.BindCommandToViper(cmd)
+		WEBSITE: https://docs.flintlock.dev/
+		
+		SUPPORT: https://github.com/weaveworks/flintlock
+		
+		`, cli.AppHelpTemplate)
 
-			if err := log.Configure(&cfg.Logging); err != nil {
-				return fmt.Errorf("configuring logging: %w", err)
-			}
+	app := cli.NewApp()
 
-			return nil
-		},
-		RunE: func(c *cobra.Command, _ []string) error {
-			return c.Help()
-		},
+	if out != nil {
+		app.Writer = out
 	}
 
-	log.AddFlagsToCommand(cmd, &cfg.Logging)
-
-	if err := addRootSubCommands(cmd, cfg); err != nil {
-		return nil, fmt.Errorf("adding subcommands: %w", err)
+	app.Name = "flintlockd"
+	app.Usage = usage
+	app.Description = `
+flintlock is a service for creating and managing the lifecycle of microVMs on a host machine. 
+Initially we will be supporting Firecracker.
+
+The primary use case for flintlock is to create microVMs on a bare-metal host where the microVMs 
+will be used as nodes in a virtualized Kubernetes cluster. It is an essential part of 
+Liquid Metal and will ultimately be driven by Cluster API Provider Microvm (coming soon).
+
+A default configuration is used if located at the default file location. It can also be set using --config flag.`
+	app.HideVersion = true
+	app.Flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:  "config,c",
+			Usage: "path to the configuration file",
+			Value: filepath.Join(defaults.ConfigurationDir, configName),
+		},
 	}
 
-	cobra.OnInitialize(initCobra)
+	log.AddFlagsToApp(app, &cfg.Logging)
+	addCommands(app, cfg)
 
-	return cmd, nil
-}
+	app.Action = func(c *cli.Context) error {
+		err := cli.ShowAppHelp(c)
+		if err != nil {
+			return err
+		}
 
-func initCobra() {
-	viper.SetEnvPrefix("FLINTLOCKD")
-	viper.AutomaticEnv()
-	viper.SetConfigType("yaml")
-	viper.SetConfigName("config")
-	viper.AddConfigPath(defaults.ConfigurationDir)
-
-	xdgCfg := os.Getenv("XDG_CONFIG_HOME")
-	if xdgCfg != "" {
-		viper.AddConfigPath("$XDG_CONFIG_HOME/flintlockd/")
-	} else {
-		viper.AddConfigPath("$HOME/.config/flintlockd/")
+		return nil
 	}
 
-	_ = viper.ReadInConfig()
-}
+	app.Before = func(context *cli.Context) error {
+		// Load the configuration file
+		configPath := context.String("config")
```

Comment:
I think we should still be using `XDG_CONFIG_HOME` as the base for the location (like we did previously)/